io/io/src/IOUtils.cxx

```diff
@@ -0,0 +1,80 @@
+#include "ROOT/IOUtils.hxx"
+
+#include "ROOT/RConfig.hxx" // R__UNIX
+
+#ifdef R__UNIX
+// getxattr
+#ifdef R__FBSD
+#include <sys/extattr.h>
+#else
+#include <sys/xattr.h>
+#endif
+
+#ifdef R__MACOSX
+/* On macOS getxattr takes two extra arguments that should be set to 0 */
+#define getxattr(path, name, value, size) getxattr(path, name, value, size, 0u, 0)
+#endif
+
+#ifdef R__FBSD
+#define getxattr(path, name, value, size) extattr_get_file(path, EXTATTR_NAMESPACE_USER, name, value, size)
+#endif
+
+#include "ROOT/StringUtils.hxx" // ROOT::StartsWith
+#include "TEnv.h"               // TEnv::GetValue
+#endif
+
+auto ROOT::Internal::GetXAttrSize(std::string_view path, std::string_view xattr)
```

Comment:
```suggestion
auto ROOT::Internal::GetXAttrSize(... path, ... xattr)
```
Unfortunately the optimization provided by `std::string_view` is not helping here.  There is no guarantees from `std::string_view` is null terminated, so with the proposed interface, one need to first turn the argument into `std::string` before passing them to `getxattr`.